src/libstore/sqlite.cc

```diff
@@ -60,6 +64,26 @@ static void traceSQL(void * x, const char * sql)
 
 SQLite::SQLite(const std::filesystem::path & path, SQLiteOpenMode mode)
 {
+// Work around a ZFS issue where SQLite's truncate() call on
+// db.sqlite-shm can randomly take up to a few seconds. See
+// https://github.com/openzfs/zfs/issues/14290#issuecomment-3074672917.
```

Comment:
```suggestion
// https://github.com/openzfs/zfs/issues/14290#issuecomment-3074672917.
// Remove this workaround when a fix is widely installed, perhaps 2027? Candidate:
// https://github.com/search?q=repo%3Aopenzfs%2Fzfs+%22Linux%3A+zfs_putpage%3A+complete+async+page+writeback+immediately%22&type=commits
```

idk about their merging / branches / rebasing, so I opted for a search URL